src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -4,38 +4,115 @@
+      return 1;
+    }
+    return 2;
   }
 
-  private boolean compliesTo(List<MIDSFields> fields, JsonNode attributes) {
-    for (var value : fields) {
-      if (attributes.get(value.getTerm()) != null) {
-        var data = attributes.get(value.getTerm()).asText();
-        if (data == null || data.trim().equals("") || data.equals("null")){
-          return false;
+  private List<MIDSFields> determineMIDSType(DigitalSpecimen digitalSpecimen) {
+    var bioType = List.of("BotanySpecimen", "MycologySpecimen", "MicrobiologyMicroOrganismSpecimen",
+        "ZoologyVertebrateSpecimen", "ZoologyInvertebrateSpecimen");
+    var paleoType = List.of("PalaeontologySpecimen");
+    var geoType = List.of("GeologyRockSpecimen", "GeologyMineralSpecimen",
+        "GeologyMixedSolidMatterSpecimen", "AstronomySpecimen");
+    if (bioType.contains(digitalSpecimen.type())) {
+      return MIDSFields.MIDS_2_BIO;
+    } else if (paleoType.contains(digitalSpecimen.type()) || geoType.contains(
+        digitalSpecimen.type())) {
+      return MIDSFields.MIDS_2_GEO_PALEO;
+    } else {
+      log.warn("Digital Specimen has unknown type: {} level 1 is highest achievable",
+          digitalSpecimen.type());
+      return Collections.emptyList();
+    }
+  }
+
+  private boolean doesNotComplyTo(List<MIDSFields> fields, JsonNode attributes) {
+    for (var field : fields) {
+      if (field.equals(MIDSFields.QUANTITATIVE_LOCATION)) {
+        if (isLackingQuantitativeLocation(attributes, field)) {
+          log.debug(MISSING_MESSAGE, field);
+          return true;
+        }
+      } else if (field.equals(MIDSFields.HAS_MEDIA)) {
+        if (isLackingMedia(attributes, field)) {
+          log.debug(MISSING_MESSAGE, field);
+          return true;
         }
       } else {
-        return false;
+        if (isFieldMissing(attributes, field)) {
+          log.debug(MISSING_MESSAGE, field);
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private boolean isFieldMissing(JsonNode attributes, MIDSFields field) {
+    for (var term : field.getTerm()) {
+      if (attributes.get(term) != null) {
+        var data = attributes.get(term).asText();
+        if (data != null && !data.trim().equals("") && !data.equals("null")) {
+          return false;
+        }
       }
     }
     return true;
   }
 
+  private boolean isLackingMedia(JsonNode attributes, MIDSFields field) {
+    for (var term : field.getTerm()) {
+      if (attributes.get(term) != null) {
+        var data = attributes.get(term).asText();
+        if (data == null || data.trim().equals("") || data.equals("null")
+            || data.equalsIgnoreCase("false")) {
+          return true;
+        }
+      } else {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean isLackingQuantitativeLocation(JsonNode attributes, MIDSFields field) {
+    for (var term : field.getTerm()) {
```

Comment:
Combined them, let me know if you think it is better. I am not completely sure but it is more concise and the is less duplication, so that might be a win.

Reply:
I think it's nicer